metrics/mean_iou/mean_iou.py

```diff
@@ -0,0 +1,237 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Mean IoU (Intersection-over-Union) metric."""
+
+import numpy as np
+
+import datasets
+
+
+_DESCRIPTION = """
+IoU is the area of overlap between the predicted segmentation and the ground truth divided by the area of union
+between the predicted segmentation and the ground truth. For binary (two classes) or multi-class segmentation,
+the mean IoU of the image is calculated by taking the IoU of each class and averaging them.
+"""
+
+_KWARGS_DESCRIPTION = """
+Args:
+    predictions: Predicted segmentation maps.
+    references: Ground truth segmentation maps.
+    num_labels (int): Number of classes (categories).
+    ignore_index (int): Index that will be ignored in evaluation.
+    nan_to_num (int, optional): If specified, NaN values will be replaced by the numbers defined by the user. Default: None.
+    label_map (dict): Mapping old labels to new labels. Default: dict().
+    reduce_zero_label (bool): Whether ignore zero label. Default: False.
+Returns:
+    dict[str, float | ndarray]:
+            <aAcc> float: Overall accuracy on all images.
+            <Acc> ndarray: Per category accuracy, shape (num_labels, ).
+            <IoU> ndarray: Per category IoU, shape (num_labels, ).
+Examples:
+
+    >>> mean_iou_metric = datasets.load_metric("mean_iou")
+    >>> results = mean_iou_metric.compute(references=[0, 1], predictions=[0, 1])
+    >>> print(results)
+"""
+
+_CITATION = """\
+authors:
+  - name: "MMSegmentation Contributors"
+title: "OpenMMLab Semantic Segmentation Toolbox and Benchmark"
+date-released: 2020-07-10
+url: "https://github.com/open-mmlab/mmsegmentation"
+license: Apache-2.0
+"""
+
+
+def f_score(precision, recall, beta=1):
+    """Calculate the f-score value.
+
+    Args:
+        precision (float): The precision value.
+        recall (float): The recall value.
+        beta (int): Determines the weight of recall in the combined score.
+    Returns:
+        [float]: The f-score value.
+    """
+    score = (1 + beta**2) * (precision * recall) / ((beta**2 * precision) + recall)
+    return score
+
+
+def intersect_and_union(pred_label, label, num_classes, ignore_index, label_map=dict(), reduce_zero_label=False):
+    """Calculate intersection and Union.
+
+    Args:
+        pred_label (ndarray):
+            Prediction segmentation map.
+        label (ndarray):
+            Ground truth segmentation map.
+        num_classes (int):
+            Number of categories.
+        ignore_index (int):
+            Index that will be ignored in evaluation.
+        label_map (dict):
+            Mapping old labels to new labels. The parameter will work only when label is str. Default: dict().
+        reduce_zero_label (bool):
+            Whether ignore zero label. The parameter will work only when label is str. Default: False.
```

Comment:
It should also be possible to make these params work when the label is a 2D array of ints, no? If not, we can have an additional "string" config that works with these params.

Reply:
Yes I actually removed the option to have label as a string (can be Numpy arrays only).